kratos/python/add_model_part_to_python.cpp

```diff
@@ -447,6 +447,93 @@ void ModelPartRemoveConditionsFromAllLevels(ModelPart& rModelPart, Flags identif
+{
+    return rModelPart.CreateNewMasterSlaveConstraint(ConstraintName, Id, rMasterDofsVector, rSlaveDofsVector, RelationMatrix, ConstantVector);
+}
+
+// Master slave constraints
+ModelPart::MasterSlaveConstraintType::Pointer CreateNewMasterSlaveConstraint2(ModelPart& rModelPart,
+                                                                                       std::string ConstraintName,
+                                                                                       ModelPart::IndexType Id,
+                                                                                       ModelPart::NodeType& rMasterNode,
+                                                                                       ModelPart::DoubleVariableType& rMasterVariable,
+                                                                                       ModelPart::NodeType& rSlaveNode,
+                                                                                       ModelPart::DoubleVariableType& rSlaveVariable,
+                                                                                       double Weight,
+                                                                                       double Constant)
+{
+    return rModelPart.CreateNewMasterSlaveConstraint(ConstraintName, Id, rMasterNode, rMasterVariable, rSlaveNode, rSlaveVariable, Weight, Constant);
+}
+
+// Master slave constraints
+ModelPart::MasterSlaveConstraintType::Pointer CreateNewMasterSlaveConstraint3(ModelPart& rModelPart,
+                                                                                        std::string ConstraintName,
+                                                                                        ModelPart::IndexType Id,
+                                                                                        ModelPart::NodeType& rMasterNode,
+                                                                                        ModelPart::VariableComponentType& rMasterVariable,
+                                                                                        ModelPart::NodeType& rSlaveNode,
+                                                                                        ModelPart::VariableComponentType& rSlaveVariable,
+                                                                                        double Weight,
+                                                                                        double Constant)
+{
+    return rModelPart.CreateNewMasterSlaveConstraint(ConstraintName, Id, rMasterNode, rMasterVariable, rSlaveNode, rSlaveVariable, Weight, Constant);
+}
+
+
+ModelPart::MasterSlaveConstraintContainerType& ModelPartGetMasterSlaveConstraints1(ModelPart& rModelPart)
+{
+    return rModelPart.MasterSlaveConstraints();
+}
+
+void ModelPartSetMasterSlaveConstraints1(ModelPart& rModelPart, ModelPart::MasterSlaveConstraintContainerType::Pointer pOtherMasterSlaveConstraints)
```

Comment:
i wonder if it would not be possible to make a perfect forwarding of the arguments to reduce the number of theses functions. Indeed construction here is more complicated than for the case of Elements and Conditions since there are multiple ways to construct the MPC (and there can be many more not contemplated here)

Reply:
@RiccardoRossi  I love the concept of perfect forwarding tried it like this 

``` 
template <typename ... Args>
ModelPart::MasterSlaveConstraintType::Pointer CreateNewMasterSlaveConstraint(ModelPart& rModelPart, Args&& ... args)
{
    return rModelPart.CreateNewMasterSlaveConstraint(std::forward<Args>(args) ...);
}
```

But unfortunately the pybind could not deduce Args and args !!

Reply:
I think that in exporting to pybind you'll have to explicitly add helper functins, however this way of using could still be useful on the c++ side